orderer/common/server/util_test.go

```diff
@@ -56,3 +66,90 @@ func TestCreateLedgerFactory(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateBootstrapBlock(t *testing.T) {
+	tempDir := t.TempDir()
+	copyYamlFiles("testdata", tempDir)
+	cryptoPath := generateCryptoMaterials(t, cryptogen, tempDir)
+	defer os.RemoveAll(cryptoPath)
+
+	systemChannelBlockPath := generateBootstrapBlock(t, tempDir, configtxgen, "system", "SampleSoloSystemChannel")
+	systemChannelBlockBytes, err := ioutil.ReadFile(systemChannelBlockPath)
+	require.NoError(t, err)
+
+	applicationChannelBlockPath := generateBootstrapBlock(t, tempDir, configtxgen, "mychannel", "SampleOrgChannel")
+	applicationChannelBlockBytes, err := ioutil.ReadFile(applicationChannelBlockPath)
+	require.NoError(t, err)
+
+	appBlock := &common.Block{}
+	require.NoError(t, proto.Unmarshal(applicationChannelBlockBytes, appBlock))
+
+	systemBlock := &common.Block{}
+	require.NoError(t, proto.Unmarshal(systemChannelBlockBytes, systemBlock))
+
+	cryptoProvider, err := sw.NewDefaultSecurityLevelWithKeystore(sw.NewDummyKeyStore())
+	require.NoError(t, err)
+
+	for _, testCase := range []struct {
+		description   string
+		block         *common.Block
+		expectedError string
+	}{
+		{
+			description:   "nil block",
+			expectedError: "nil block",
+		},
+		{
+			description:   "empty block",
+			block:         &common.Block{},
+			expectedError: "empty block data",
+		},
+		{
+			description: "bad envelope",
+			block: &common.Block{
+				Data: &common.BlockData{
+					Data: [][]byte{{1, 2, 3}},
+				},
+			},
+			expectedError: "failed extracting envelope from block",
+		},
+		{
+			description:   "application channel block",
+			block:         appBlock,
+			expectedError: "the block isn't a system channel block because it lacks ConsortiumsConfig",
+		},
+		{
+			description: "system channel block",
+			block:       systemBlock,
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			err := ValidateBootstrapBlock(testCase.block, cryptoProvider)
+			if testCase.expectedError == "" {
+				require.NoError(t, err)
+				return
+			}
+			require.Error(t, err)
+			require.Contains(t, err.Error(), testCase.expectedError)
+		})
+	}
+}
+
+func generateBootstrapBlock(t *testing.T, tempDir, configtxgen, channel, profile string) string {
+	gt := NewGomegaWithT(t)
+	// create a genesis block for the specified channel and profile
+	genesisBlockPath := filepath.Join(tempDir, channel+".block")
+	cmd := exec.Command(
+		configtxgen,
+		"-channelID", channel,
+		"-profile", profile,
+		"-outputBlock", genesisBlockPath,
+		"--configPath", tempDir,
+	)
+	configtxgenProcess, err := gexec.Start(cmd, nil, nil)
+	gt.Expect(err).NotTo(HaveOccurred())
+	gt.Eventually(configtxgenProcess, time.Minute).Should(gexec.Exit(0))
+	gt.Expect(configtxgenProcess.Err).To(gbytes.Say("Writing genesis block"))
+
+	return genesisBlockPath
+}
```

Comment:
This is one way to generate a genesis block, but we don't need it in this package, as you can use 
```
// produces a system channel genesis file to make sure the server detects it and refuses to start
func produceGenesisFileEtcdRaft(t *testing.T, channelID string, tmpDir string) (string, []byte)
```
which already exists in this file

Reply:
You can also use this one:
```
func produceGenesisFileEtcdRaftAppChannel(...)
```